Correct Sample Message Behaviour

momo3404 opened the pull request:
In the changes made to the request feedback behaviour [here](https://github.com/portagenetwork/roadmap/pull/1064/files), the sample message was incorrectly changed to reflect the organisation's current feedback message. The correct behaviour is for the sample message to simply serve as a static example to org admins.

Changes proposed in this PR:
- This PR makes the sample message static and also edits the naming of the functions in `orgs_helper` for more clarity.
- The function previously used to display the feedback message in both the text box and sample message (`display_feedback_message`) will be renamed to `display_editable_feedback_message` as it is only used for the editable text box message.
- The function `display_sample_feedback_message` is used to display the static sample feedback message.

momo3404 committed:
Add password complexity validation and error messages

momo3404 committed:
Add password_checklist css and dmp-assistant stylesheet block

momo3404 committed:
Add password_checklist html

momo3404 committed:
Add password_checklist javascript

momo3404 committed:
Fix breaking tests

momo3404 committed:
Add password complexity and validation tests

momo3404 committed:
Correct sample message behaviour and rename functions

momo3404 committed:
Correct sample message behaviour and rename functions

momo3404 committed:
Add Changelog entry

momo3404 committed:
Add Changelog entry

aaronskiba commented on app/helpers/orgs_helper.rb:
```diff
@@ -33,6 +34,25 @@ def display_feedback_message(org, current_user, feedback_message, plan_name = ni
+  # current_user - The current user we're showing feedback message to
+  #
+  # Returns String
+  def display_sample_feedback_message(org, current_user)
```
Should we instead reuse `FeedbacksHelper#feedback_confirmation_default_message` (`app/helpers/feedbacks_helper.rb`) here?

momo3404 replied:
Yes that would definitely be cleaner

momo3404 committed:
Reuse FeedbacksHelper message

momo3404 committed:
Reuse FeedbacksHelper message

momo3404 committed:
Rename password-requirements to password-checklist

momo3404 committed:
Refactor user_spec tests

dependabot[bot] committed:
chore(deps): bump docker/metadata-action from 3 to 5

Bumps [docker/metadata-action](https://github.com/docker/metadata-action) from 3 to 5.
- [Release notes](https://github.com/docker/metadata-action/releases)
- [Commits](https://github.com/docker/metadata-action/compare/v3...v5)

---
updated-dependencies:
- dependency-name: docker/metadata-action
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump docker/login-action from 1.10.0 to 3.5.0

Bumps [docker/login-action](https://github.com/docker/login-action) from 1.10.0 to 3.5.0.
- [Release notes](https://github.com/docker/login-action/releases)
- [Commits](https://github.com/docker/login-action/compare/f054a8b539a109f9f41c372932f1ae047eff08c9...184bdaa0721073962dff0199f1fb9940f07167d1)

---
updated-dependencies:
- dependency-name: docker/login-action
  dependency-version: 3.5.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump docker/setup-buildx-action from 2 to 3

Bumps [docker/setup-buildx-action](https://github.com/docker/setup-buildx-action) from 2 to 3.
- [Release notes](https://github.com/docker/setup-buildx-action/releases)
- [Commits](https://github.com/docker/setup-buildx-action/compare/v2...v3)

---
updated-dependencies:
- dependency-name: docker/setup-buildx-action
  dependency-version: '3'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

momo3404 committed:
Merge pull request #1142 from portagenetwork/dependabot/github_actions/integration/docker/metadata-action-5

chore(deps): bump docker/metadata-action from 3 to 5

momo3404 committed:
Merge pull request #1140 from portagenetwork/dependabot/github_actions/integration/docker/login-action-3.5.0

chore(deps): bump docker/login-action from 1.10.0 to 3.5.0

momo3404 committed:
Merge pull request #1139 from portagenetwork/dependabot/github_actions/integration/docker/setup-buildx-action-3

chore(deps): bump docker/setup-buildx-action from 2 to 3

aaronskiba commented on app/helpers/orgs_helper.rb:
```diff
@@ -33,6 +36,22 @@ def display_feedback_message(org, current_user, feedback_message, plan_name = ni
+    email = org.contact_email || EMAIL_PLACEHOLDER
+    username = current_user.name(false) || USER_PLACEHOLDER
+
+    # feedback_confirmation_default_message represents the default feedback message
```
Maybe this is partly just my own preference, but I wonder if the comments are a little overly thorough sometimes?
 - `# feedback_confirmation_default_message represents the default feedback message ...`
 - `# current_user - The current user we're showing feedback message to`

I like comments like `feedback_confirmation_default_message belongs to FeedbacksHelper` that help explain non-obvious things. But I think Ruby and Rails are often very expressive/readable on their own.

momo3404 replied:
Yea you're right, I tend to go overboard with comments sometimes. I'll rework these to be more concise.

momo3404 committed:
Rework comments to be more concise

momo3404 committed:
Rework comments to be more concise

momo3404 committed:
Add Changelog Entry

aaronskiba committed:
Add `const` to `passwordRequirements`

aaronskiba committed:
Wrap passwordChecklist.js in function / enable early return

- This change adds the early return of `passwordChecklist.js` when a page doesn't include any relevant elements (i.e. `if (passwordInputs.length === 0 || !passwordRequirements) return;`).
- To enable this early return, we are now assigning the prior code to the `passwordChecklist` function, and then calling it immediately.

aaronskiba committed:
Update passwordChecklist.js to reduce DOM queries

Add `ruleIconsMap` to map password rules to their corresponding `<i>` icons.
- Eliminates repeated DOM queries previously being performed in `updateRequirements()`

aaronskiba committed:
Simplify passwordChecklist.js to single password input

Replaced the `passwordsInputs` array with a single `passwordInput` variable. Pages never have more than one password field that uses `passwordField.js`, and this change should simplify the code a little bit.

aaronskiba committed:
Rename variables in `passwordChecklist.js`

- Renamed `passwordRequirements` to `passwordChecklist` to reflect elements actual name
- Renamed `passwordChecklist` function name to `initPasswordChecklist` to better distinguish it from the newly added `passwordChecklist` variable
- Renamed `updateRequirements` to `updateChecklistIcons` to clarify behaviour

momo3404 committed:
Correct password validation in password reset

aaronskiba committed:
Add comments to `passwordChecklist.js`

aaronskiba committed:
Remove redundant explicit `password_field` IDs

- Rails automatically assigns `id="user_password"` to password fields, so
explicit `id: 'user_password'` attributes were unnecessary.

- Additionally, the manually defined `id: 'user_password'` in `app/views/shared/_create_account_form.html.erb` had no effect. Rather, `id="new_user_password"` was being generated due to `namespace: 'new'` being specified in the form.

aaronskiba committed:
Merge pull request #1162 from portagenetwork/aaron/passwordChecklistJS-updates

`passwordChecklist.js` Changes

momo3404 committed:
Merge pull request #1158 from portagenetwork/momo/password-policy

Improve Password Policy

momo3404 committed:
Merge branch 'momo/sample-message-fix' of https://github.com/portagenetwork/roadmap into momo/sample-message-fix
